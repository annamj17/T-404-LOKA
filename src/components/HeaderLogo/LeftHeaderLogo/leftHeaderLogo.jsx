import React from 'react';
import { View, Image, TouchableOpacity } from 'react-native';

import logoImage from '../../../../assets/K3imaginelogo.png';
import styles from '../styles';

const HeaderImage = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => {
          navigation.navigate('HomeView');
        }}
      >
        <Image style={styles.logo} source={logoImage} resizeMode="contain" />
      </TouchableOpacity>
    </View>
  );
};

export default HeaderImage;
