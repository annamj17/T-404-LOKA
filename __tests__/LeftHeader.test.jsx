import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import LeftHeaderLogo from '../src/components/HeaderLogo/LeftHeaderLogo/leftHeaderLogo';

describe('Should render the LeftHeader page without crashing', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<LeftHeaderLogo />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

describe('LeftHeader tests', () => {
  const onNavigateSpy = jest.fn();
  const props = {
    navigation: {
      navigate: onNavigateSpy
    }
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LeftHeaderLogo {...props} />);
  });

  it('Should find 1 TouchableOpacity', () => {
    expect(wrapper.find('TouchableOpacity')).toHaveLength(1);
  });

  it('should test onPress navigation', () => {
    const press = wrapper.find('TouchableOpacity');
    press.simulate('Press');
    expect(props.navigation.navigate).toHaveBeenCalledWith('HomeView');
  });
});
