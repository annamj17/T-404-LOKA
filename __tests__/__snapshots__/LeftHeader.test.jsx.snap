// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the LeftHeader page without crashing renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": "100%",
          "width": 160,
        }
      }
    />
  </View>
</View>
`;
